jina/executors/devices.py

```diff
@@ -22,11 +22,6 @@ def device(self):
             please use the environment variable `CUDA_VISIBLE_DEVICES`.
         """
 
-    @property
-    @abstractmethod
-    def is_gpu(self) -> bool:
```

Comment:
I think keeping this is a safer way to have this `mixin`. You force anyone who uses it to define it. By just using `self.on_gpu` there is a risk that you are mixing it with a class that does not provide this method. So it affects its independence.

Reply:
`on_gpu` is a meta tag available to all executors. it indicates whether user wants to put it on gpu. That's it. no guarantee it will be put on to gpu.

Reply:
oh btw, all docstring inside `executors.metas` are *not* being picked up to our docs!